src/sync/impl/sync_file.cpp

```diff
@@ -200,7 +201,15 @@ std::string create_timestamped_template(const std::string& prefix, int wildcard_
     wildcard_count = std::min(WILDCARD_MAX, std::max(WILDCARD_MIN, wildcard_count));
     std::time_t time = std::time(nullptr);
     std::stringstream stream;
-    stream << prefix << "-" << std::put_time(std::localtime(&time), "%Y%m%d-%H%M%S") << "-" << std::string(wildcard_count, 'X');
+    stream << prefix << "-";
+#if __GNUC__ < 5
```

Comment:
No, not really. It's a GCC thing:

```kneth@blackbird:~$ uname -a
Linux blackbird 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux
kneth@blackbird:~$ g++ --version
g++ (Debian 4.9.2-10) 4.9.2
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

kneth@blackbird:~$ cat pt.cpp
#include <sstream>
#include <iomanip>
#include <iostream>
#include <ctime>

int main(int argc, char *argv[]) {
    std::stringstream ss;
    std::time_t time = std::time(nullptr);
    ss << std::put_time(std::localtime(&time), "%Y%m%d-%H%M%S");
    std::cout << ss.str() << std::endl;
}
kneth@blackbird:~$ g++ pt.cpp
pt.cpp: In function 'int main(int, char**)':
pt.cpp:8:34: error: 'nullptr' was not declared in this scope
     std::time_t time = std::time(nullptr);
                                  ^
pt.cpp:9:11: error: 'put_time' is not a member of 'std'
     ss << std::put_time(std::localtime(&time), "%Y%m%d-%H%M%S");```

Compiling on Debian GNU/Linux (stable) gives the same error.